src/wp-includes/block-supports/generated-classname.php

```diff
@@ -57,6 +57,11 @@ function wp_apply_generated_classname_support( $block_type ) {
 		if ( $block_classname ) {
 			$attributes['class'] = $block_classname;
 		}
+
+		$variation = infer_block_variation( $block_type, $block_attributes );
+		if ( $variation ) {
+			$attributes['class'] .= ' ' . $block_classname . '-' . $variation;
+		}
```

Comment:
Might make more sense to reuse the classname generation function here for consistency.

```php
$block_classname = wp_get_block_default_classname( $block_type->name );

$variation = infer_block_variation( $block_type, $block_attributes );
if ( $variation ) {
	$block_classname .= ' ' . wp_get_block_default_classname( $block_type->name . '/' . $variation->name );
}

if ( $block_classname ) {
	$attributes['class'] = $block_classname;
}
```

Reply:
https://github.com/WordPress/wordpress-develop/pull/6594/commits/e9c03e64e0efcc98440bea7083e8f2780571c4bf